src/utils/modul-date/modul-date.ts

```diff
@@ -299,34 +299,35 @@ export default class ModulDate {
 
         // Otherwise we try to build the date from a partial date string (2010-12-01 or 2010/12/01)
 
-
-        const parts: string[] = DATE_FORMAT_REGEX.exec(value) as string[];
-        if (!parts || parts.length < 4) {
+        if (!DATE_FORMAT_REGEX.test(value)) {
```

Comment:
On devrait peut être utiliser Date.parse() ici au lieu de faire tout la poutine a bras avec un regex qui introduit une certaine complexité et possiblement des bug (car généralement une regex = 4 bugs :neckbeard: )

Par example, dans date-format on fait :
`!isNaN(Date.parse(control.value)`

Reply:
Lorsque je mets `isNaN(Date.parse(control.value)` plusieurs tests unitaires plantent.

`ModulDate › initialization › from a string › of a supported format › then a date with format '18/1/1' is a date comparable with date '2018-01-01'`

`ModulDate › initialization › from a string › of a supported format › then a date with format '18/1/18' is a date comparable with date '2018-01-18'`

`A single date state › when calling event (from abstract class) › month events › month-select › when selecting a month after the maximum date, the current month will be set to the maximum`

`A range date state › when calling event (from abstract class) › month events › month-select › when selecting a month after the maximum date, the current month will be set to the maximum`

Reply:
et si tu fais isNaN(new Date(control.value))?

Reply:
Ok, je suis d'accord que ça sort du scope de cette PR mais on peut garder ça en tête si modul-date nous cause des problèmes